dashboard/src/t5gweb/t5gweb.py

```diff
@@ -136,6 +137,80 @@ def get_new_comments():
 
     return accounts
 
+def get_trending_cards():
+        # Set the default configuration values
+    cfg = set_cfg()
+    cfg['query'] = "case_summary:*webscale* OR case_tags:*shift_telco5g* OR case_summary:*cnv,* OR case_tags:*cnv*"    
+    try:
+        conn = jira.JIRA(cfg['options'], basic_auth=(cfg['jira_user'], cfg['jira_pass']))
+    except jira.exceptions as e:
+        if e.status_code == 401:
+            print("Login to JIRA failed. Check your username and password")
+            exit (1)
+    board = libtelco5g.get_board_id(conn, cfg['board'])
+    sprint = libtelco5g.get_latest_sprint(conn, board.id, cfg['sprintname'])
+    cards = conn.search_issues('component = "KNI Labs & Field" AND (project = KNIECO OR project = KNIP AND issuetype = Epic AND status != Obsolete) AND labels = "Trends" AND updated >= "2021-07-01" AND updated <= "2021-09-30" ORDER BY Rank ASC', maxResults=1000)
+    token=libtelco5g.get_token(cfg['offline_token'])
+
+    cases_json=libtelco5g.get_cases_json(token,cfg['query'],cfg['fields'], exclude_closed= False)
+    cases=libtelco5g.get_cases(cases_json, include_tags=True)
+
+
+    cards_dict = {}
+    for card in cards:
+        cards_dict[card.key] = None
+
+    #Associate each card with its corresponding case number
+    for card_id in cards_dict:
+        links = conn.remote_links(card_id)
+        for link in links:
+            t = conn.remote_link(card_id, link)
+            if t.raw['object']['title'] == "Support Case":
+                t_case_number = libtelco5g.get_case_number(t.raw['object']['url'])
+                if len(t_case_number) > 0:
+                    cards_dict[card_id] = t_case_number
+
+    # Get rid of cards with no Support Case Link
+    linked_cards = {card: case for card, case in cards_dict.items() if case is not None}
+
+    time_now = datetime.now(timezone.utc)
+
+    # Add other details to dictionary, like case number and comments on card that were made in the last seven days
+    detailed_cards= {}
+    for card_name in linked_cards:
```

Comment:
would it make sense to turn some of this in to functions? bit of duplicated code here.